### PR TITLE
Add executed test tasks to build table

### PR DESCRIPTION
**Pull request by blindpirate**
So we can find out unexpected cache misses more easily.

### CODE REVIEW

File: ge-export/src/main/kotlin/org/gradle/devprod/collector/enterprise/export/extractor/extractors.kt [hunk 1]
```diff
@@ -89,6 +89,22 @@ object FirstTestTaskStart : SingleEventExtractor<Pair<String, Instant>?>("TaskSt
...
 }
 
+// https://docs.gradle.com/enterprise/event-model-javadoc/com/gradle/scan/eventmodel/gradle/TaskFinished_1_0.html
+object ExecutedTestTasks : SingleEventExtractor<List<String>>("TaskFinished") {
+    override fun extract(events: Iterable<BuildEvent>): List<String> {
+        return events
+            .filter { it.data?.stringProperty("outcome") in listOf("SUCCESS", "FAILED") }
+            .flatMap {
+                val path = it.data?.stringProperty("path")
+                if (path?.endsWith("Test") == true) {
```
Comment (wolfs) [inline ge-export/src/main/kotlin/org/gradle/devprod/collector/enterprise/export/extractor/extractors.kt:99]: You could use `filter` here as well and filter for tasks which end in `Test`.

File: persistence/src/main/resources/db/migration/V1.14__Add_executed_test_tasks_array_to_build.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+alter table build
+    add executed_test_tasks varchar(255) not null default '{}'
```
Comment (wolfs) [inline persistence/src/main/resources/db/migration/V1.14__Add_executed_test_tasks_array_to_build.sql:2]:
```suggestion
    add executed_test_tasks varchar(255) [] not null default '{}'
```

Isn't the `[]` missing here?

File: persistence/src/main/java/org/gradle/devprod/collector/persistence/generated/jooq/tables/records/BuildRecord.java [hunk 3]
```diff
@@ -510,6 +534,11 @@ public Boolean value16() {
         return getBuildCacheStoreFailure();
     }
 
+    @Override
+    public String value17() {
```
Comment (wolfs) [inline persistence/src/main/java/org/gradle/devprod/collector/persistence/generated/jooq/tables/records/BuildRecord.java:538]: This should be `String[]`, right?

**Review by wolfs — CHANGES_REQUESTED** [review]
Looks like the type is `String` instead of `String[]`.

File: ge-export/src/main/kotlin/org/gradle/devprod/collector/enterprise/export/extractor/extractors.kt (hunk 1 above)
Comment (blindpirate) [reply ge-export/src/main/kotlin/org/gradle/devprod/collector/enterprise/export/extractor/extractors.kt]: Now I use a `mapNotNull` and `filter`.

File: persistence/src/main/resources/db/migration/V1.14__Add_executed_test_tasks_array_to_build.sql (hunk 2 above)
Comment (blindpirate) [reply persistence/src/main/resources/db/migration/V1.14__Add_executed_test_tasks_array_to_build.sql]: Yeah... the `[]` was lost mistakenly when I added `not null` clause.

**Review by wolfs — APPROVED** [review]
LGTM! Let's see if we need to add some kind of index later on.

**Pull request merged**